### PR TITLE
josema/ch86072/improve-metrics-in-case-of-on-prem

### PR DESCRIPTION
**Pull request by josemazo**
CH story [ch86072](https://app.clubhouse.io/cartoteam/story/86072/improve-metrics-in-case-of-on-prem)

### CODE REVIEW

File: cartoframes/utils/metrics.py [hunk 1]
```diff
@@ -69,15 +78,16 @@ def check_valid_metrics_uuid(metrics_config):
...
     metrics_data = {
         'event_version': EVENT_VERSION,
         'event_time': get_local_time(),
         'event_source': EVENT_SOURCE,
         'event_name': event_name,
         'source_version': __version__,
         'installation_id': get_metrics_uuid(),
-        'runtime_env': get_runtime_env()
+        'runtime_env': get_runtime_env(),
+        'api_used': CLOUD_API if server_domain_tld in [PROD_DOMAIN_TLD, STAG_DOMAIN_TLD] else CUSTOM_API
```
Comment (Jesus89) [inline cartoframes/utils/metrics.py:90]: I would move this logic to `utils.get_api_used(server_domain_tld)` or `get_api_used(server_domain_tld)` (this file)

File: cartoframes/utils/metrics.py [hunk 2]
```diff
@@ -87,9 +97,10 @@ def build_metrics_data(event_name, extra_metrics_data):
 
 
 @silent_fail
-def post_metrics(event_name, extra_metrics_data):
-    json_data = build_metrics_data(event_name, extra_metrics_data)
-    result = requests.post('https://bmetrics.cartodb.net', json=json_data, timeout=2)
+def post_metrics(event_name, extra_metrics_data, server_domain_tld):
+    metrics_server = STAG_METRICS_SERVER if server_domain_tld == STAG_DOMAIN_TLD else PROD_METRICS_SERVER
```
Comment (Jesus89) [inline cartoframes/utils/metrics.py:101]: I would move this logic to `utils.get_metrics_server(server_domain_tld)` or `get_metrics_server(server_domain_tld)` (this file)

File: cartoframes/utils/metrics.py [hunk 3]
```diff
@@ -100,22 +111,32 @@ def wrapper_func(*args, **kwargs):
...
+
     try:
-        credentials = get_parameter_from_decorator(
-            'credentials', decorated_function, *args, **kwargs)
+        credentials = get_parameter_from_decorator('credentials', decorated_function, *args, **kwargs)
         credentials = get_credentials(credentials)
-        return {'user_id': credentials.user_id} if credentials and credentials.user_id else {}
+
+        hostname = urlparse(credentials.base_url).hostname
+        server_domain_tld = '.'.join(hostname.split('.')[-2:])
```
Comment (Jesus89) [inline cartoframes/utils/metrics.py:131]: Move this to a function `utils.get_server_domain_tld(base_url)` or `get_metrics_server(base_url)` (this file)

**Review by Jesus89 — CHANGES_REQUESTED** [review]
LGTM!

I suggested moving some code to utility functions to improve readability

File: cartoframes/utils/metrics.py (hunk 1 above)
Comment (josemazo) [reply cartoframes/utils/metrics.py:90]: :raised_hands:

File: cartoframes/utils/metrics.py (hunk 3 above)
Comment (josemazo) [reply cartoframes/utils/metrics.py]: :rocket:

**Comment by josemazo** [comment]
Re CR, please!!

**Review by Jesus89 — APPROVED** [review]
:+1:

**Pull request merged**